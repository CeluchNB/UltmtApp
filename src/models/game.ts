import { DisplayUser } from '../types/user'
import { Realm } from '@realm/react'
import { Tournament } from '../types/tournament'
import { CreateGame, Game, GameStatus, PointStats } from '../types/game'
import { DisplayTeam, GuestTeam } from '../types/team'

export class GameSchema {
    static schema: Realm.ObjectSchema = {
        name: 'Game',
        primaryKey: '_id',
        properties: {
            _id: 'string',
            creator: 'DisplayUser',
            teamOne: 'DisplayTeam',
            teamTwo: 'GuestTeam',
            teamTwoDefined: 'bool',
            scoreLimit: 'int',
            halfScore: 'int',
            startTime: 'date',
            softcapMins: 'int',
            hardcapMins: 'int',
            playersPerPoint: 'int',
            timeoutPerHalf: 'int',
            floaterTimeout: 'bool',
            tournament: 'Tournament?',
            teamOneScore: 'int',
            teamTwoScore: 'int',
            teamOnePlayers: 'DisplayUser[]',
            teamTwoPlayers: 'DisplayUser[]',
            resolveCode: 'string',
            statsPoints: 'PointStats[]',
            offline: 'bool',
            teamOneStatus: 'string',
            teamTwoStatus: 'string',
        },
    }

    _id: string
    creator: DisplayUser
    teamOne: DisplayTeam
    teamTwo: GuestTeam
    teamTwoDefined: boolean
    scoreLimit: number
    halfScore: number
    startTime: Date
    softcapMins: number
    hardcapMins: number
    playersPerPoint: number
    timeoutPerHalf: number
    floaterTimeout: boolean
    tournament?: Tournament
    teamOneScore: number
    teamTwoScore: number
    teamOnePlayers: DisplayUser[]
    teamTwoPlayers: DisplayUser[]
    resolveCode: string
    statsPoints: PointStats[]
    offline: boolean
    teamOneStatus: GameStatus
    teamTwoStatus: GameStatus

    constructor(
        game: Game,
        offline: boolean = false,
        statsPoints: PointStats[] = [],
    ) {
        this._id = game._id
        this.creator = game.creator
        this.teamOne = game.teamOne
        this.teamTwo = game.teamTwo
        this.teamTwoDefined = game.teamTwoDefined
        this.scoreLimit = game.scoreLimit
        this.halfScore = game.halfScore
        this.startTime = game.startTime
        this.softcapMins = game.softcapMins
        this.hardcapMins = game.hardcapMins
        this.playersPerPoint = game.playersPerPoint
        this.timeoutPerHalf = game.timeoutPerHalf
        this.floaterTimeout = game.floaterTimeout
        this.tournament = game.tournament
        this.teamOneScore = game.teamOneScore
        this.teamTwoScore = game.teamTwoScore
        this.teamOnePlayers = game.teamOnePlayers
        this.teamTwoPlayers = game.teamTwoPlayers
        this.resolveCode = game.resolveCode || ''
        this.statsPoints = statsPoints
        this.offline = offline
        this.teamOneStatus = game.teamOneStatus
        this.teamTwoStatus = game.teamTwoStatus
    }

    static createOfflineGame(game: CreateGame, teamOnePlayers: DisplayUser[]) {
        return {
            _id: new Realm.BSON.ObjectID().toHexString(),
            creator: game.creator,
            teamOne: game.teamOne,
            teamTwo: game.teamTwo,
            teamTwoDefined: game.teamTwoDefined,
            scoreLimit: game.scoreLimit,
            halfScore: game.halfScore,
            startTime: new Date(game.startTime),
            softcapMins: game.softcapMins,
            hardcapMins: game.hardcapMins,
            playersPerPoint: game.playersPerPoint,
            timeoutPerHalf: game.timeoutPerHalf,
            floaterTimeout: game.floaterTimeout,
            tournament: game.tournament,
            teamOneScore: 0,
            teamTwoScore: 0,
            teamOneActive: true,
            teamTwoActive: false,
            teamOnePlayers: teamOnePlayers,
            teamTwoPlayers: [],
            resolveCode: '',
            statsPoints: [],
            offline: true,
        }
    }
}
export const PointStatsSchema: Realm.ObjectSchema = {
    name: 'PointStats',
    embedded: true,
    properties: {
        _id: 'string',
        pointStats: 'PlayerPointStats[]',
    },
}

export const PlayerPointStatsSchema: Realm.ObjectSchema = {
    name: 'PlayerPointStats',
    embedded: true,
    properties: {
        _id: 'string',
        firstName: 'string',
        lastName: 'string',
        username: 'string?',
        pointsPlayed: 'int',
        goals: 'int',
        assists: 'int',
        turnovers: 'int',
        blocks: 'int',
    },
}
