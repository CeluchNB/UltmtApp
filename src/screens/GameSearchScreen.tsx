import { Game } from '../types/game'
import GameCard from '../components/atoms/GameCard'
import { GameSearchProps } from '../types/navigation'
import MapSection from '../components/molecules/MapSection'
import SearchBar from '../components/atoms/SearchBar'
import { searchGames } from '../services/data/game'
import React, { useEffect } from 'react'
import { SafeAreaView, ScrollView, StyleSheet, View } from 'react-native'
import { useColors, useLazyData } from '../hooks'

const GameSearchScreen: React.FC<GameSearchProps> = ({ route }) => {
    const { colors } = useColors()
    const { live } = route.params

    const { data, loading, fetch } = useLazyData<Game[]>(searchGames)

    useEffect(() => {
        fetch('', live)
        // eslint-disable-next-line react-hooks/exhaustive-deps
    }, [])

    const search = (q?: string) => {
        fetch(q, live)
    }

    const styles = StyleSheet.create({
        screen: {
            height: '100%',
            backgroundColor: colors.primary,
        },
        container: {
            width: '90%',
            alignSelf: 'center',
        },
    })

    return (
        <SafeAreaView style={styles.screen}>
            <View>
                <SearchBar
                    placeholder="Search games..."
                    onChangeText={search}
                    filter={true}
                    onFilterPress={() => {
                        console.log('filter press')
                    }}
                />
            </View>
            <ScrollView style={styles.container}>
                <MapSection
                    title="Results"
                    showButton={false}
                    showCreateButton={false}
                    loading={loading}
                    listData={data}
                    renderItem={game => (
                        <GameCard
                            key={game._id}
                            game={game}
                            onPress={() => {}}
                        />
                    )}
                />
            </ScrollView>
        </SafeAreaView>
    )
}

export default GameSearchScreen
