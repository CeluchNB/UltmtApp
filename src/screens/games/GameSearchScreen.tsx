import { Chip } from 'react-native-paper'
import FilterModal from '../../components/molecules/FilterModal'
import { Game } from '../../types/game'
import GameCard from '../../components/atoms/GameCard'
import { GameSearchProps } from '../../types/navigation'
import SearchBar from '../../components/atoms/SearchBar'
import { parseLiveValue } from '../../utils/form-utils'
import { searchGames } from '../../services/data/game'
import { useColors } from '../../hooks'
import { useInfiniteQuery } from 'react-query'
import { weight } from '../../theme/fonts'
import {
    ActivityIndicator,
    FlatList,
    SafeAreaView,
    ScrollView,
    StyleSheet,
    Text,
    View,
} from 'react-native'
import React, { useState } from 'react'

const PAGE_SIZE = 10

const GameSearchScreen: React.FC<GameSearchProps> = ({ route }) => {
    const { colors } = useColors()
    const { live: defaultLive } = route.params

    const [modalVisible, setModalVisible] = useState(false)
    const [q, setQ] = useState('')
    const [live, setLive] = useState(defaultLive)
    const [after, setAfter] = useState(new Date('2022-01-02'))
    const [before, setBefore] = useState(new Date())

    const { data, isLoading, isFetchingNextPage, fetchNextPage } =
        useInfiniteQuery(
            ['games', q, live, after, before],
            ({ pageParam = 0 }): Promise<Game[]> => {
                return searchGames(
                    q,
                    parseLiveValue(live),
                    after.toLocaleDateString('en-US'),
                    before.toLocaleDateString('en-US'),
                    PAGE_SIZE,
                    PAGE_SIZE * pageParam,
                )
            },
            {
                getNextPageParam: (lastPage, allPages) => {
                    if (lastPage.length > 0) {
                        return allPages.length
                    }
                    return undefined
                },
            },
        )

    const search = (query: string = '') => {
        setQ(query)
    }

    const loadMore = () => {
        if (!isFetchingNextPage && !isLoading) {
            fetchNextPage()
        }
    }

    const closeFilters = (formData: {
        live: string
        after: Date
        before: Date
    }) => {
        const {
            live: filterLive,
            after: filterAfter,
            before: filterBefore,
        } = formData
        setLive(filterLive)
        setAfter(filterAfter)
        setBefore(filterBefore)
        setModalVisible(false)
    }

    const styles = StyleSheet.create({
        screen: {
            height: '100%',
            backgroundColor: colors.primary,
        },
        container: {
            width: '90%',
            alignSelf: 'center',
            marginBottom: 5,
        },
        chip: {
            borderRadius: 4,
            marginRight: 5,
            backgroundColor: colors.textPrimary,
            color: colors.primary,
        },
        resultChip: {
            borderRadius: 4,
            marginRight: 5,
            backgroundColor: colors.primary,
            color: colors.textPrimary,
            borderColor: colors.textPrimary,
        },
        chipText: {
            fontWeight: weight.bold,
            color: colors.primary,
        },
        resultChipText: {
            color: colors.textPrimary,
            fontWeight: weight.bold,
        },
    })

    return (
        <SafeAreaView style={styles.screen}>
            <View>
                <SearchBar
                    placeholder="Search games..."
                    onChangeText={search}
                    filter={true}
                    onFilterPress={() => {
                        setModalVisible(true)
                    }}
                />
            </View>
            <FilterModal
                visible={modalVisible}
                defaultValues={{ live, after, before }}
                onClose={closeFilters}
            />
            <View>
                <ScrollView
                    style={styles.container}
                    horizontal={true}
                    showsHorizontalScrollIndicator={false}>
                    <View>
                        <Chip mode="outlined" style={styles.resultChip}>
                            <Text style={styles.resultChipText}>
                                {data?.pages.flat().length} results
                            </Text>
                        </Chip>
                    </View>
                    <View>
                        <Chip mode="outlined" style={styles.resultChip}>
                            <Text style={styles.resultChipText}>
                                {
                                    data?.pages
                                        .flat()
                                        .filter(g => g.teamOneActive).length
                                }{' '}
                                live results
                            </Text>
                        </Chip>
                    </View>
                    <View>
                        <Chip mode="flat" style={styles.chip}>
                            {live === 'true' && (
                                <Text style={styles.chipText}>Live games</Text>
                            )}
                            {live === 'false' && <Text>Completed games</Text>}
                            {live !== 'true' && live !== 'false' && (
                                <Text style={styles.chipText}>All games</Text>
                            )}
                        </Chip>
                    </View>
                    <View>
                        <Chip mode="outlined" style={styles.chip}>
                            <Text style={styles.chipText}>
                                After: {after.toLocaleDateString()}
                            </Text>
                        </Chip>
                    </View>
                    <View>
                        <Chip mode="flat" style={styles.chip}>
                            <Text style={styles.chipText}>
                                Before: {before.toLocaleDateString('en-US')}
                            </Text>
                        </Chip>
                    </View>
                </ScrollView>
            </View>
            <FlatList
                data={data?.pages.flat()}
                renderItem={({ item: game }) => (
                    <GameCard key={game._id} game={game} onPress={() => {}} />
                )}
                keyExtractor={(item, i) => {
                    return `${item._id}${i}`
                }}
                onEndReached={loadMore}
                onEndReachedThreshold={0.1}
                ListFooterComponent={() => {
                    return isLoading || isFetchingNextPage ? (
                        <ActivityIndicator
                            color={colors.textPrimary}
                            size="large"
                            testID="infinite-scroll-indicator"
                        />
                    ) : null
                }}
                testID="game-search-list"
            />
        </SafeAreaView>
    )
}

export default GameSearchScreen
