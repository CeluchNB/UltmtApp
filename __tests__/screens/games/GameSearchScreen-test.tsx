import * as GameServices from '../../../src/services/network/game'
import { GameSearchProps } from '../../../src/types/navigation'
import GameSearchScreen from '../../../src/screens/games/GameSearchScreen'
import { NavigationContainer } from '@react-navigation/native'
import React from 'react'
import { game } from '../../../fixtures/data'
import { QueryClient, QueryClientProvider } from 'react-query'
import {
    act,
    fireEvent,
    render,
    waitForElementToBeRemoved,
} from '@testing-library/react-native'

import MockDate from 'mockdate'
MockDate.set('01 October 2022 00:00 UTC')

jest.mock('react-native/Libraries/Animated/NativeAnimatedHelper')
jest.mock('../../../src/components/atoms/GameCard', () => () => {
    return <div>Game</div>
})

const props: GameSearchProps = {
    navigation: {} as any,
    route: { params: { live: 'true' } } as any,
}
const client = new QueryClient()

beforeAll(() => {
    jest.useFakeTimers()
})

beforeEach(() => {
    jest.clearAllMocks()
    jest.spyOn(GameServices, 'searchGames').mockReturnValueOnce(
        Promise.resolve({
            data: { games: [game] },
            status: 200,
            statusText: 'Good',
            headers: {},
            config: {},
        }),
    )
})

afterAll(() => {
    jest.useRealTimers()
})

it('should match snapshot', async () => {
    const snapshot = render(
        <NavigationContainer>
            <QueryClientProvider client={client}>
                <GameSearchScreen {...props} />
            </QueryClientProvider>
        </NavigationContainer>,
    )

    await waitForElementToBeRemoved(() =>
        snapshot.getByTestId('infinite-scroll-indicator'),
    )

    expect(snapshot.toJSON()).toMatchSnapshot()
})

it('should search', async () => {
    const spy = jest.spyOn(GameServices, 'searchGames').mockReturnValueOnce(
        Promise.resolve({
            data: { games: [] },
            status: 200,
            statusText: 'Good',
            headers: {},
            config: {},
        }),
    )
    const { getByPlaceholderText, queryByText } = render(
        <NavigationContainer>
            <QueryClientProvider client={client}>
                <GameSearchScreen {...props} />
            </QueryClientProvider>
        </NavigationContainer>,
    )

    const searchBar = getByPlaceholderText('Search games...')
    fireEvent.changeText(searchBar, 'text')
    await act(async () => {})

    const result = await queryByText('Game')
    expect(result).toBeNull()

    expect(spy).toHaveBeenCalled()
})

it('should use filter modal', async () => {
    const spy = jest.spyOn(GameServices, 'searchGames').mockReturnValueOnce(
        Promise.resolve({
            data: { games: [] },
            status: 200,
            statusText: 'Good',
            headers: {},
            config: {},
        }),
    )

    const { getByText, queryByText } = render(
        <NavigationContainer>
            <QueryClientProvider client={client}>
                <GameSearchScreen {...props} />
            </QueryClientProvider>
        </NavigationContainer>,
    )

    const searchBar = getByText('Filter')
    fireEvent.press(searchBar)

    const doneButton = getByText('done')
    fireEvent.press(doneButton)
    await act(async () => {})

    const result = await queryByText('Game')
    expect(result).toBeNull()

    expect(spy).toHaveBeenCalled()
})

it('should load more on scroll', async () => {
    const spy = jest.spyOn(GameServices, 'searchGames').mockReturnValueOnce(
        Promise.resolve({
            data: { games: [game] },
            status: 200,
            statusText: 'Good',
            headers: {},
            config: {},
        }),
    )

    const { getByTestId } = render(
        <NavigationContainer>
            <QueryClientProvider client={client}>
                <GameSearchScreen {...props} />
            </QueryClientProvider>
        </NavigationContainer>,
    )

    const scrollView = getByTestId('game-search-list')
    fireEvent.scroll(scrollView, {
        nativeEvent: {
            contentSize: { height: 100, width: 100 },
            contentOffset: { y: 10, x: 0 },
            layoutMeasurement: { height: 100, width: 100 }, // Dimensions of the device
        },
    })
    await act(async () => {})

    expect(spy).toHaveBeenCalledTimes(2)
})
