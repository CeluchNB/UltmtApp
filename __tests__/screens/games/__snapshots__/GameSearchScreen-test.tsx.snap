// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should match snapshot 1`] = `
<RCTSafeAreaView
  emulateUnlessSupported={true}
  style={
    Object {
      "backgroundColor": "#303030",
      "height": "100%",
    }
  }
>
  <View>
    <View
      style={
        Object {
          "alignSelf": "center",
          "display": "flex",
          "flexDirection": "row",
          "width": "90%",
        }
      }
    >
      <View
        style={
          Array [
            Object {
              "backgroundColor": "rgb(231, 231, 231)",
              "borderTopLeftRadius": 4,
              "borderTopRightRadius": 4,
            },
            Object {
              "backgroundColor": "#303030",
              "color": "#ffd600",
              "flex": 1,
              "marginBottom": 5,
            },
          ]
        }
      >
        <View
          collapsable={false}
          style={
            Object {
              "backgroundColor": "#ffd600",
              "bottom": 0,
              "height": 2,
              "left": 0,
              "position": "absolute",
              "right": 0,
              "transform": Array [
                Object {
                  "scaleY": 0.5,
                },
              ],
              "zIndex": 1,
            }
          }
        />
        <View
          style={
            Array [
              Object {
                "paddingBottom": 0,
                "paddingTop": 0,
              },
              Object {
                "minHeight": 64,
              },
            ]
          }
        >
          <TextInput
            editable={true}
            focusable={true}
            multiline={false}
            onBlur={[Function]}
            onChangeText={[Function]}
            onFocus={[Function]}
            onPressOut={[Function]}
            placeholder="Search games..."
            placeholderTextColor="#ababab"
            selectionColor="#ffd600"
            style={
              Array [
                Object {
                  "flexGrow": 1,
                  "margin": 0,
                },
                Object {
                  "paddingLeft": 12,
                  "paddingRight": 12,
                },
                Object {
                  "height": 64,
                },
                Object {
                  "paddingBottom": 0,
                  "paddingTop": 0,
                },
                Object {
                  "color": "#ffd600",
                  "fontFamily": "System",
                  "fontSize": 41.0958904109589,
                  "fontWeight": undefined,
                  "textAlign": "left",
                  "textAlignVertical": "center",
                },
                false,
                Array [
                  Object {},
                ],
              ]
            }
            underlineColorAndroid="transparent"
            value=""
          />
        </View>
      </View>
      <View
        collapsable={false}
        style={
          Object {
            "alignSelf": "center",
            "backgroundColor": "transparent",
            "borderColor": "#ffd600",
            "borderRadius": 4,
            "borderStyle": "solid",
            "borderWidth": 0.5,
            "elevation": 0,
            "marginStart": 5,
            "minWidth": 64,
          }
        }
      >
        <View
          accessibilityRole="button"
          accessibilityState={
            Object {
              "disabled": false,
            }
          }
          accessible={true}
          focusable={true}
          onClick={[Function]}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            Array [
              Object {
                "overflow": "hidden",
              },
              Object {
                "borderRadius": 4,
              },
            ]
          }
        >
          <View
            style={
              Array [
                Object {
                  "alignItems": "center",
                  "flexDirection": "row",
                  "justifyContent": "center",
                },
                undefined,
              ]
            }
          >
            <Text
              numberOfLines={1}
              selectable={false}
              style={
                Array [
                  Object {
                    "color": "#000000",
                    "fontFamily": "System",
                    "fontWeight": "400",
                  },
                  Object {
                    "textAlign": "left",
                  },
                  Array [
                    Object {
                      "letterSpacing": 1,
                      "marginHorizontal": 16,
                      "marginVertical": 9,
                      "textAlign": "center",
                    },
                    undefined,
                    Object {
                      "textTransform": "uppercase",
                    },
                    Object {
                      "color": "#ffd600",
                      "fontFamily": "System",
                      "fontWeight": "500",
                    },
                    Object {
                      "fontFamily": "System",
                      "fontWeight": "500",
                    },
                    undefined,
                  ],
                ]
              }
            >
              Filter
            </Text>
          </View>
        </View>
      </View>
    </View>
  </View>
  <Modal
    animationType="slide"
    hardwareAccelerated={false}
    onRequestClose={[Function]}
    style={Object {}}
    testID="filter-modal"
    transparent={true}
    visible={false}
  />
  <View>
    <RCTScrollView
      horizontal={true}
      showsHorizontalScrollIndicator={false}
      style={
        Object {
          "alignSelf": "center",
          "marginBottom": 5,
          "width": "90%",
        }
      }
    >
      <View>
        <View>
          <View
            collapsable={false}
            style={
              Object {
                "backgroundColor": "#303030",
                "borderColor": "#ffd600",
                "borderRadius": 4,
                "borderStyle": "solid",
                "borderWidth": 0.5,
                "color": "#ffd600",
                "elevation": 0,
                "flexDirection": "column",
                "marginRight": 5,
              }
            }
          >
            <View
              accessibilityRole="button"
              accessibilityState={
                Object {
                  "disabled": true,
                  "selected": false,
                }
              }
              accessible={true}
              focusable={false}
              onClick={[Function]}
              onResponderGrant={[Function]}
              onResponderMove={[Function]}
              onResponderRelease={[Function]}
              onResponderTerminate={[Function]}
              onResponderTerminationRequest={[Function]}
              onStartShouldSetResponder={[Function]}
              style={
                Array [
                  Object {
                    "overflow": "hidden",
                  },
                  Array [
                    Object {
                      "borderRadius": 4,
                    },
                    Object {
                      "flexGrow": 1,
                    },
                  ],
                ]
              }
            >
              <View
                style={
                  Array [
                    Object {
                      "alignItems": "center",
                      "flexDirection": "row",
                      "flexGrow": 1,
                      "paddingLeft": 4,
                      "position": "relative",
                    },
                    Object {
                      "paddingRight": 4,
                    },
                  ]
                }
              >
                <Text
                  numberOfLines={1}
                  selectable={false}
                  style={
                    Array [
                      Object {
                        "color": "#000000",
                        "fontFamily": "System",
                        "fontWeight": "400",
                      },
                      Object {
                        "textAlign": "left",
                      },
                      Array [
                        Object {
                          "lineHeight": 24,
                          "marginVertical": 4,
                          "minHeight": 24,
                          "textAlignVertical": "center",
                        },
                        Object {
                          "color": "rgba(0, 0, 0, 0.87)",
                          "fontFamily": "System",
                          "fontWeight": "400",
                          "marginLeft": 8,
                          "marginRight": 8,
                        },
                        undefined,
                      ],
                    ]
                  }
                >
                  <Text
                    style={
                      Object {
                        "color": "#ffd600",
                        "fontWeight": "bold",
                      }
                    }
                  >
                    1
                     results
                  </Text>
                </Text>
              </View>
            </View>
          </View>
        </View>
        <View>
          <View
            collapsable={false}
            style={
              Object {
                "backgroundColor": "#303030",
                "borderColor": "#ffd600",
                "borderRadius": 4,
                "borderStyle": "solid",
                "borderWidth": 0.5,
                "color": "#ffd600",
                "elevation": 0,
                "flexDirection": "column",
                "marginRight": 5,
              }
            }
          >
            <View
              accessibilityRole="button"
              accessibilityState={
                Object {
                  "disabled": true,
                  "selected": false,
                }
              }
              accessible={true}
              focusable={false}
              onClick={[Function]}
              onResponderGrant={[Function]}
              onResponderMove={[Function]}
              onResponderRelease={[Function]}
              onResponderTerminate={[Function]}
              onResponderTerminationRequest={[Function]}
              onStartShouldSetResponder={[Function]}
              style={
                Array [
                  Object {
                    "overflow": "hidden",
                  },
                  Array [
                    Object {
                      "borderRadius": 4,
                    },
                    Object {
                      "flexGrow": 1,
                    },
                  ],
                ]
              }
            >
              <View
                style={
                  Array [
                    Object {
                      "alignItems": "center",
                      "flexDirection": "row",
                      "flexGrow": 1,
                      "paddingLeft": 4,
                      "position": "relative",
                    },
                    Object {
                      "paddingRight": 4,
                    },
                  ]
                }
              >
                <Text
                  numberOfLines={1}
                  selectable={false}
                  style={
                    Array [
                      Object {
                        "color": "#000000",
                        "fontFamily": "System",
                        "fontWeight": "400",
                      },
                      Object {
                        "textAlign": "left",
                      },
                      Array [
                        Object {
                          "lineHeight": 24,
                          "marginVertical": 4,
                          "minHeight": 24,
                          "textAlignVertical": "center",
                        },
                        Object {
                          "color": "rgba(0, 0, 0, 0.87)",
                          "fontFamily": "System",
                          "fontWeight": "400",
                          "marginLeft": 8,
                          "marginRight": 8,
                        },
                        undefined,
                      ],
                    ]
                  }
                >
                  <Text
                    style={
                      Object {
                        "color": "#ffd600",
                        "fontWeight": "bold",
                      }
                    }
                  >
                    1
                     
                    live results
                  </Text>
                </Text>
              </View>
            </View>
          </View>
        </View>
        <View>
          <View
            collapsable={false}
            style={
              Object {
                "backgroundColor": "#ffd600",
                "borderColor": "#ffd600",
                "borderRadius": 4,
                "borderStyle": "solid",
                "borderWidth": 0.5,
                "color": "#303030",
                "elevation": 0,
                "flexDirection": "column",
                "marginRight": 5,
              }
            }
          >
            <View
              accessibilityRole="button"
              accessibilityState={
                Object {
                  "disabled": true,
                  "selected": false,
                }
              }
              accessible={true}
              focusable={false}
              onClick={[Function]}
              onResponderGrant={[Function]}
              onResponderMove={[Function]}
              onResponderRelease={[Function]}
              onResponderTerminate={[Function]}
              onResponderTerminationRequest={[Function]}
              onStartShouldSetResponder={[Function]}
              style={
                Array [
                  Object {
                    "overflow": "hidden",
                  },
                  Array [
                    Object {
                      "borderRadius": 4,
                    },
                    Object {
                      "flexGrow": 1,
                    },
                  ],
                ]
              }
            >
              <View
                style={
                  Array [
                    Object {
                      "alignItems": "center",
                      "flexDirection": "row",
                      "flexGrow": 1,
                      "paddingLeft": 4,
                      "position": "relative",
                    },
                    Object {
                      "paddingRight": 4,
                    },
                  ]
                }
              >
                <Text
                  numberOfLines={1}
                  selectable={false}
                  style={
                    Array [
                      Object {
                        "color": "#000000",
                        "fontFamily": "System",
                        "fontWeight": "400",
                      },
                      Object {
                        "textAlign": "left",
                      },
                      Array [
                        Object {
                          "lineHeight": 24,
                          "marginVertical": 4,
                          "minHeight": 24,
                          "textAlignVertical": "center",
                        },
                        Object {
                          "color": "rgba(0, 0, 0, 0.87)",
                          "fontFamily": "System",
                          "fontWeight": "400",
                          "marginLeft": 8,
                          "marginRight": 8,
                        },
                        undefined,
                      ],
                    ]
                  }
                >
                  <Text
                    style={
                      Object {
                        "color": "#303030",
                        "fontWeight": "bold",
                      }
                    }
                  >
                    Live games
                  </Text>
                </Text>
              </View>
            </View>
          </View>
        </View>
        <View>
          <View
            collapsable={false}
            style={
              Object {
                "backgroundColor": "#ffd600",
                "borderColor": "rgba(0, 0, 0, 0.29)",
                "borderRadius": 4,
                "borderStyle": "solid",
                "borderWidth": 0.5,
                "color": "#303030",
                "elevation": 0,
                "flexDirection": "column",
                "marginRight": 5,
              }
            }
          >
            <View
              accessibilityRole="button"
              accessibilityState={
                Object {
                  "disabled": true,
                  "selected": false,
                }
              }
              accessible={true}
              focusable={false}
              onClick={[Function]}
              onResponderGrant={[Function]}
              onResponderMove={[Function]}
              onResponderRelease={[Function]}
              onResponderTerminate={[Function]}
              onResponderTerminationRequest={[Function]}
              onStartShouldSetResponder={[Function]}
              style={
                Array [
                  Object {
                    "overflow": "hidden",
                  },
                  Array [
                    Object {
                      "borderRadius": 4,
                    },
                    Object {
                      "flexGrow": 1,
                    },
                  ],
                ]
              }
            >
              <View
                style={
                  Array [
                    Object {
                      "alignItems": "center",
                      "flexDirection": "row",
                      "flexGrow": 1,
                      "paddingLeft": 4,
                      "position": "relative",
                    },
                    Object {
                      "paddingRight": 4,
                    },
                  ]
                }
              >
                <Text
                  numberOfLines={1}
                  selectable={false}
                  style={
                    Array [
                      Object {
                        "color": "#000000",
                        "fontFamily": "System",
                        "fontWeight": "400",
                      },
                      Object {
                        "textAlign": "left",
                      },
                      Array [
                        Object {
                          "lineHeight": 24,
                          "marginVertical": 4,
                          "minHeight": 24,
                          "textAlignVertical": "center",
                        },
                        Object {
                          "color": "rgba(0, 0, 0, 0.87)",
                          "fontFamily": "System",
                          "fontWeight": "400",
                          "marginLeft": 8,
                          "marginRight": 8,
                        },
                        undefined,
                      ],
                    ]
                  }
                >
                  <Text
                    style={
                      Object {
                        "color": "#303030",
                        "fontWeight": "bold",
                      }
                    }
                  >
                    After: 
                    1/2/2022
                  </Text>
                </Text>
              </View>
            </View>
          </View>
        </View>
        <View>
          <View
            collapsable={false}
            style={
              Object {
                "backgroundColor": "#ffd600",
                "borderColor": "#ffd600",
                "borderRadius": 4,
                "borderStyle": "solid",
                "borderWidth": 0.5,
                "color": "#303030",
                "elevation": 0,
                "flexDirection": "column",
                "marginRight": 5,
              }
            }
          >
            <View
              accessibilityRole="button"
              accessibilityState={
                Object {
                  "disabled": true,
                  "selected": false,
                }
              }
              accessible={true}
              focusable={false}
              onClick={[Function]}
              onResponderGrant={[Function]}
              onResponderMove={[Function]}
              onResponderRelease={[Function]}
              onResponderTerminate={[Function]}
              onResponderTerminationRequest={[Function]}
              onStartShouldSetResponder={[Function]}
              style={
                Array [
                  Object {
                    "overflow": "hidden",
                  },
                  Array [
                    Object {
                      "borderRadius": 4,
                    },
                    Object {
                      "flexGrow": 1,
                    },
                  ],
                ]
              }
            >
              <View
                style={
                  Array [
                    Object {
                      "alignItems": "center",
                      "flexDirection": "row",
                      "flexGrow": 1,
                      "paddingLeft": 4,
                      "position": "relative",
                    },
                    Object {
                      "paddingRight": 4,
                    },
                  ]
                }
              >
                <Text
                  numberOfLines={1}
                  selectable={false}
                  style={
                    Array [
                      Object {
                        "color": "#000000",
                        "fontFamily": "System",
                        "fontWeight": "400",
                      },
                      Object {
                        "textAlign": "left",
                      },
                      Array [
                        Object {
                          "lineHeight": 24,
                          "marginVertical": 4,
                          "minHeight": 24,
                          "textAlignVertical": "center",
                        },
                        Object {
                          "color": "rgba(0, 0, 0, 0.87)",
                          "fontFamily": "System",
                          "fontWeight": "400",
                          "marginLeft": 8,
                          "marginRight": 8,
                        },
                        undefined,
                      ],
                    ]
                  }
                >
                  <Text
                    style={
                      Object {
                        "color": "#303030",
                        "fontWeight": "bold",
                      }
                    }
                  >
                    Before: 
                    10/1/2022
                  </Text>
                </Text>
              </View>
            </View>
          </View>
        </View>
      </View>
    </RCTScrollView>
  </View>
  <RCTScrollView
    ListFooterComponent={[Function]}
    data={
      Array [
        Object {
          "_id": "game1",
          "creator": Object {
            "_id": "user1",
            "firstName": "First 1",
            "lastName": "Last 1",
            "username": "first1last1",
          },
          "floaterTimeout": true,
          "halfScore": 8,
          "hardcapMins": 90,
          "playersPerPoint": 7,
          "points": Array [],
          "resolveCode": "111111",
          "scoreLimit": 15,
          "softcapMins": 75,
          "startTime": 2022-10-12T00:00:00.000Z,
          "teamOne": Object {
            "_id": "team1",
            "name": "Temper",
            "place": "Pittsburgh",
            "seasonEnd": "2022",
            "seasonStart": "2022",
            "teamname": "pghtemper",
          },
          "teamOneActive": true,
          "teamOnePlayers": Array [],
          "teamOneScore": 3,
          "teamTwo": Object {
            "name": "Sockeye",
          },
          "teamTwoActive": false,
          "teamTwoDefined": false,
          "teamTwoPlayers": Array [],
          "teamTwoScore": 0,
          "timeoutPerHalf": 1,
          "tournament": Object {
            "_id": "tourney1",
            "endDate": 2022-10-23T00:00:00.000Z,
            "eventId": "nationals22",
            "name": "Club Nationals 2022",
            "startDate": 2022-10-20T00:00:00.000Z,
          },
        },
      ]
    }
    getItem={[Function]}
    getItemCount={[Function]}
    keyExtractor={[Function]}
    onContentSizeChange={[Function]}
    onEndReached={[Function]}
    onEndReachedThreshold={0.1}
    onLayout={[Function]}
    onMomentumScrollBegin={[Function]}
    onMomentumScrollEnd={[Function]}
    onScroll={[Function]}
    onScrollBeginDrag={[Function]}
    onScrollEndDrag={[Function]}
    removeClippedSubviews={false}
    renderItem={[Function]}
    scrollEventThrottle={50}
    stickyHeaderIndices={Array []}
    testID="game-search-list"
    viewabilityConfigCallbackPairs={Array []}
  >
    <View>
      <View
        onLayout={[Function]}
        style={null}
      >
        <div>
          Game
        </div>
      </View>
      <View
        onLayout={[Function]}
      />
    </View>
  </RCTScrollView>
</RCTSafeAreaView>
`;
